src/spell/modules/spell/spell/src/kv/collection.rs

```diff
@@ -85,6 +92,34 @@ pub fn list_get_strings(key: &str) -> StringListValue {
     result.into()
 }
 
+#[marine]
+/// Remove a value from a list of strings. If the value is in several places, remove all of them.
+/// `list_order` is used to preserve order of elements in a list, not the actual index.
+pub fn list_remove_value(key: &str, value: &str) -> UnitValue {
+    let conn = db();
+    let result: eyre::Result<()> = try {
+        let mut statement = conn.prepare(
+            r#"
+            DELETE FROM kv
+                WHERE key = ?
+                  AND string = ?
+                  AND list_order != -1
+            RETURNING list_order
+        "#,
+        )?;
+        statement.bind(1, key)?;
+        statement.bind(2, value)?;
+        let list: Vec<i64> = fetch_rows(statement, |statement| {
+            Ok(Some(statement.read::<i64>(0)?))
+        });
+        if list.len() == 0 {
+            return Err(eyre::eyre!("value not found"))?;
```

Comment:
What value? How to act on this error?

Reply:
This means the `list_remove_value` doesn't remove anything since it wasn't found. `UnitValue` doesn't have an `absent` flag to set (nor needs it). I can return `U32Value` with the number of entries removed or introduce RemoveResult, but it feels excessive.

Reply:
So, I changed it so list_remove wouldn't complain when it didn't remove anything. I guess when can change it if we need to react smartly to it later.